src/battery-check/battery-check.c

```diff
@@ -0,0 +1,141 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "battery-util.h"
+#include "build.h"
+#include "constants.h"
+#include "errno-util.h"
+#include "glyph-util.h"
+#include "fd-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "socket-util.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "Checks battery level to see whether there's enough charge.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static void battery_check_send_plymouth_message(char *message) {
+        assert(message);
+
+        ssize_t w;
+        static const union sockaddr_union sa = PLYMOUTH_SOCKET;
+        _cleanup_close_ int fd = -EBADF;
+        _cleanup_free_ char *plymouth_message = NULL, *plymouth_shutdown = NULL;
+
+        if (asprintf(&plymouth_message, "M\002%c%s", (int) strlen(message), message) < 0) {
+                log_oom();
+                return;
+        }
+        if (asprintf(&plymouth_shutdown, "C\002%c%s", (int) strlen("shutdown"), "shutdown") < 0) {
+                log_oom();
```

Comment:
you can do this in one line with `return (void) log_oom();`

Reply:
It sort of failed build locally when i tried. Maybe it was something else I missed.

Reply:
Certainly sir.